pkg/await/states/types.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package states
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-kubernetes/pkg/logging"
+)
+
+// Result specifies the result of a Condition applied to an input object.
```

Comment:
Might want to mention that this is a `v1.PodCondition`, and write a bit more about what this is actually _for_ -- _e.g._, it's the `Result` of a check that a `Pod` has some `v1.PodCondition` that is true, with some metadata useful for rendering messages about it.

Reply:
This actually isn't specific to `PodCondition`s; it's a generic condition that is used to determine readiness of the relevant resource.